plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/expression/AggregateFunctionPatterns.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.query.expression;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.matching.Captures;
+import io.trino.matching.Match;
+import io.trino.matching.Pattern;
+import io.trino.matching.PatternVisitor;
+import io.trino.matching.Property;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public final class AggregateFunctionPatterns
+{
+    // Extracted from io.trino.plugin.jdbc.expression.AggregateFunctionPatterns
```

Comment:
Let's move `AggregateFunctionPatterns` to plugin-toolkit instead of copying it.

Reply:
https://github.com/trinodb/trino/pull/8578

Reply:
@elonazoulay Was there any conclusion here? I agree it'd be much nicer to avoid a copy to prevent code from drifting away.

Reply:
I was wondering if I could make RewriteContext extendible, or add the groupingColumns to it. Count distinct in pinot needs to verify that the input column is contained in the grouping columns if there is a passthrough query. Let me know what you think.

Reply:
Agreed, would be much better to use #8578 .